src/CSV.jl

```diff
@@ -43,13 +43,13 @@ type Options
     escapechar::UInt8
     separator::UInt8
     decimal::UInt8
-    null::@compat(String) # how null is represented in the dataset
+    null::Compat.ASCIIString # how null is represented in the dataset
     nullcheck::Bool   # do we have a custom null value to check for
     dateformat::Dates.DateFormat
     datecheck::Bool   # do we have a custom dateformat to check for
 end
 
-Options(;delim=COMMA,quotechar=QUOTE,escapechar=ESCAPE,null::@compat(String)="",dateformat=Dates.ISODateFormat) =
+Options(;delim=COMMA,quotechar=QUOTE,escapechar=ESCAPE,null::Compat.ASCIIString="",dateformat=Dates.ISODateFormat) =
```

Comment:
Given that no other argument has a type here, just remove it and leave the constructor do the conversion automatically.